libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1099,6 +1129,71 @@ void WaveClip::Resample(int rate, BasicUI::ProgressDialog *progress)
    }
 }
 
+void WaveClip::ApplyStretchRatio()
+{
+   const auto stretchRatio = GetStretchRatio();
+   if (stretchRatio == 1.0)
+      return;
+
+   auto success = false;
+   auto newSequences = GetEmptySequenceCopies();
+
+   Finally Do { [&, trimLeftBeforeStretch = mTrimLeft,
+                 trimRightBeforeStretch = mTrimRight] {
+      // Whether successful or not, the right thing to do is to restore the
+      // original trim values.
+      this->SetTrimLeft(trimLeftBeforeStretch);
+      this->SetTrimRight(trimRightBeforeStretch);
+      if (success)
+      {
+         this->mClipStretchRatio = 1.0;
+         this->mRawAudioTempo = this->mProjectTempo;
+         assert(this->GetStretchRatio() == 1.0);
+      }
+      else
+         std::swap(mSequences, newSequences);
+   } };
+
+   SetTrimLeft(0);
+   SetTrimRight(0);
+
+   constexpr auto durationToDiscard = 0.0;
+   constexpr auto blockSize = 1024;
```

Comment:
Did experimentation with larger numbers might find an easy way to improve performance?